### PR TITLE
refactor(Authoring): Create project authoring lesson component

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- Created project authoring lesson component

## Test
- In the project authoring view make sure
  - lessons show up properly
  - lessons can be clicked to go into them
  - lesson checkboxes work properly for moving and deleting lessons

### CODE REVIEW

File: src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    aria-label="Select lesson"
+    i18n-aria-label
+  >
+  </mat-checkbox>
+  <div
+    class="projectItemTitleDiv"
+    fxLayoutAlign="start center"
+    fxLayoutGap="10px"
+    (click)="nodeClicked(item.key)"
+  >
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.html:16]: Rename?
```suggestion
    (click)="setCurrentNode(item.key)"
```

File: src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<div id="{{ item.key }}" fxLayoutAlign="start center" class="pointer groupHeader">
+  <mat-checkbox
+    color="primary"
+    [(ngModel)]="item.checked"
+    (click)="selectNode()"
+    [disabled]="disableSelection"
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.html:6]: Emit event directly?
```suggestion
    (click)="selectNodeEvent.emit()"
```

File: src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+})
+export class ProjectAuthoringLessonComponent {
+  @Input() disableSelection: boolean = false;
+  @Input() item: any;
+  @Output() selectNodeEvent: EventEmitter<void> = new EventEmitter<void>();
+
+  constructor(private dataService: TeacherDataService) {}
+
+  nodeClicked(nodeId: string): void {
+    this.dataService.setCurrentNodeByNodeId(nodeId);
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring-lesson/project-authoring-lesson.component.ts:18]: Add access modifier?
```suggestion
  protected setCurrentNode(nodeId: string): void {
```

**Comment by hirokiterashima** [comment]
Functionality works well for the lessons in Active section of the AT unit view. Looks like you forgot to convert the lessons in the Inactive section?

I added some code comments inline for consideration. Also, remember to generate messages.xlf.

**Comment by geoffreykwan** [comment]
I changed it so unused lessons now uses the new component. I also made the minor changes you pointed out.

**Review by hirokiterashima — APPROVED** [review]
Behavior LGTM for both active and inactive lessons.

Can you address this issue before merging:

It looks like you'll need to modify the new component so that we pass in an event to ```ProjectAuthoringComponent.selectNode($event: Event)```

Right now, each time I select a lesson, I see this error in the js console:
```
project-authoring-lesson.component.html:5 ERROR TypeError: Cannot read properties of undefined (reading 'stopPropagation')
    at ProjectAuthoringComponent.selectNode (project-authoring.component.ts:197:12)
    at ProjectAuthoringComponent_ng_container_22_div_1_Template_project_authoring_lesson_selectNodeEvent_1_listener (project-authoring.component.html:142:30)
```

**Pull request merged**